wolfSSH/src-ja/chapter13.md

```diff
@@ -0,0 +1,1475 @@
+
+
+- WS_CBIO_ERR_GENERAL (-1): 一般的な予期しないエラー 
+- WS_CBIO_ERR_WANT_READ (-2): ソケットの読み取りブロック（再度リードせよ）
+- WS_CBIO_ERR_WANT_WRITE (-2): ソケットの書き込みブロック（再度ライトせよ）
+- WS_CBIO_ERR_CONN_RST (-3): コネクションがリセットされた
+- WS_CBIO_ERR_ISR (-4): 割り込み発生
+- WS_CBIO_ERR_CONN_CLOSE (-5): コネクションがクローンした
+- WS_CBIO_ERR_TIMEOUT (-6): ソケットタイムアウト
+
+##  初期化 /シャットダウン
+
+
+
+### wolfSSH_Init()
+
+
+
+**用法**
+
+**説明**
+
+wolfSSHライブラリを初期化します。アプリケーションごとに1回、ライブラリへの他の呼び出しの前に呼び出される必要があります。
+
+**戻り値**
+
+WS_SUCCESS
+WS_CRYPTO_FAILED
+
+**引数**
+
+なし
+
+**関連項目**
+
+wolfSSH_Cleanup()
+
+```
+#include <wolfssh/ssh.h>
+int wolfSSH_Init(void);
+```
+
+### wolfSSH_Cleanup()
+
+
+
+**用法**
+
+**説明**
+
+使用完了したら、wolfSSHライブラリをクリーンアップします。アプリケーションの終了前に呼び出される必要があります。本関数呼び出し後は、ライブラリAPIの呼び出しはできません。
+
+**戻り値**
+
+**WS_SUCCESS**
+
+**WS_CRYPTO_FAILED**
+
+**引数**
+
+なし
+
+**関連項目**
+
+wolfSSH_Init()
+
+```
+#include <wolfssh/ssh.h>
+int wolfSSH_Cleanup(void);
+```
+
+##  デバッグ出力関数
+
+
+
+### wolfSSH_Debugging_ON()
+
+
+
+**用法**
+
+**説明**
+
+実行中にデバッグロギングを有効にします。ビルド時にデバッグが無効になっている場合、何もしません。
+
+
+**戻り値**
+
+なし
+
+**引数**
+
+なし
+
+**関連項目**
+
+wolfSSH_Debugging_OFF()
+
+```
+#include <wolfssh/ssh.h>
+void wolfSSH_Debugging_ON(void);
+```
+
+### wolfSSH_Debugging_OFF()
+
+
+
+**用法**
+
+**説明**
+
+実行時にデバッグロギングを無効にします。ビルド時にデバッグが無効になっている場合、何もしません。
+
+
+**戻り値**
+
+なし
+
+**引数**
+
+なし
+
+**関連項目**
+
+wolfSSH_Debugging_ON()
+
+```
+#include <wolfssh/ssh.h>
+void wolfSSH_Debugging_OFF(void);
+```
+
+##  コンテキスト関数
+
+
+
+### wolfSSH_CTX_new()
+
+
+
+**用法**
+
+**説明**
+
+wolfSSHコンテキストオブジェクトを作成します。このオブジェクトはwolfSSHセッションオブジェクトのファクトリとして使用されます。
+
+**戻り値**
+
+**WOLFSSH_CTX** – 割り当てられたWOLFSSH_CTXオブジェクトへのポインターあるいはNULL
+
+**引数**
+
+**side** – クライアントサイド(実装なし)またはサーバーサイドを示します<br>
+**heap** – メモリ割り当てに使用するヒープへのポインター
+
+**関連項目**
+
+wolfSSH_wolfSSH_CTX_free()
+
+```
+#include <wolfssh/ssh.h>
+WOLFSSH_CTX* wolfSSH_CTX_new(byte side , void* heap );
+```
+
+### wolfSSH_CTX_free()
+
+
+
+**用法**
+
+**説明**
+
+WOLFSSH_CTXオブジェクトを解放します
+
+**戻り値**
+
+なし
+
+**引数**
+
+**ctx** – WOLFSSH_CTXオブジェクト
+
+**関連項目**
+
+wolfSSH_wolfSSH_CTX_new()
+
+```
+#include <wolfssh/ssh.h>
+void wolfSSH_CTX_free(WOLFSSH_CTX* ctx );
+```
+
+### wolfSSH_CTX_SetBanner()
+
+
+**用法**
+
+**説明**
+
+バナーメッセージをセットする
+
+**戻り値**
+
+WS_BAD_ARGUMENT
+WS_SUCCESS
+
+**引数**
+
+**ssh** - wolfSSHオブジェクト<br>
+**newBanner** - バナーメッセージ文字列
+
+```
+#include <wolfssh/ssh.h>
+int wolfSSH_CTX_SetBanner(WOLFSSH_CTX* ctx , const char* newBanner );
+```
+
+### wolfSSH_CTX_UsePrivateKey_buffer()
+
+
+**用法**
+
+**説明** <br>
+この関数は、秘密鍵バッファをSSHコンテキストにロードします。ファイルの代わりにバッファーを入力として呼び出されます。バッファは、**insz** の **in** 引数によって提供されます。<br>
+
+**引数** 
+
+**format** バッファのタイプを指定します：**wolfssh_format_asn1** または **wolfssl_format_pem** (現時点では未実装)。
+
+
+**戻り値**
+
+**WS_SUCCESS** <br>
+**WS_BAD_ARGUMENT** – 少なくとも一つの引数が不正<br>
+**WS_BAD_FILETYPE_E** – フォーマットが不正<br>
+**WS_UNIMPLEMENTED_E** – PEMフォーマットは未対応<br>
+**WS_MEMORY_E** – メモリ確保エラー<br>
+**WS_RSA_E** – RSA鍵をデコードできない<br>
+**WS_BAD_FILE_E** – バッファを解析できない<br>
+
```

Comment:
NL are not working. Please review Return value of other functions as well.

Reply:
Added new lines after <br>s so that work in PDF.